server/superdesk/publish/publish_content.py

```diff
@@ -63,8 +63,16 @@ def publish(subscriber, destination):
 def transmit_items(queue_items, subscriber, destination):
     failed_items = []
 
+    output_channels = superdesk.get_resource_service('output_channels').get(req=None, lookup={})
```

Comment:
Reason for getting all output_channels is to avoid multiple calls to Mongo. We can get only active channels if the number of output_channels are more.

If we need gracefulness then yes, I do think it's better to fetch channels in the previous function and start the thread.